### ISSUE TITLE
[Bug] Drag and drop with multi-directional scrollable behavior 

### ISSUE THREAD

**Issue by tsquillario** [karvulf/flutter-reorderable-grid-view#74]
The idea is that the GridView of cards is too large for the screen so I have 2 `SingleChildScrollView`, one for each direction. It's best to test with a phone rather than a tablet. There are 2 issues.

1. The `enableScrollingWhileDragging` doesn't seem to work
2. The drag and drop reordering doesn't work after you scroll the window. In the gif below you can see it works at first, then after you scroll the window, the drag and drop position is off.

I tried moving the `SingleChildScrollView` to the top after the `body`, same result. Portrait and Landscape orientation produces the same result. Tried it with `InteractiveViewer` as well and that seems to exhibit buddy behavior as well. 

Am I doing something wrong? Thanks!!!!

Updated my example repo.
https://github.com/tsquillario/flutter-reorderable-grid-view-example1

![untitled](https://user-images.githubusercontent.com/333300/228352449-16de2c12-fa87-4141-9bd6-73aefa2947b7.gif)

**Comment by karvulf** [comment]
Hi @tsquillario 
Thanks for opening the issue. I never used it that way that I had 2 `SingleChildScrollView`. I have to check if that is even possible with my code. I think currently I just use the scroll position of one of your `SingleChildScrollView` which means that the drag and drop would be buggy in one scroll direction if you have two. If I can get access to both, then I think it should be possible to drag and drop in both scrolling directions.
I will check that this week, possibly on the week-end.

**Comment by tsquillario** [comment]
Found another potential issue. I added a zoom control to resize the GridView. When you zoom in or out the size of the card changes but the drag preview doesn't. 

Create a separate branch for `InteractiveViewer` I think that may be the better widget to use for this.

https://github.com/tsquillario/flutter-reorderable-grid-view-example1/tree/interactiveviewer

![untitled2](https://user-images.githubusercontent.com/333300/229151154-5becc414-c09f-4166-b381-143dc352a242.gif)

**Comment by karvulf** [comment]
Yeah the main issue is that I always expected that you return a `GridView` directly and not any other widgets like `InteractiveViewer`. So that's the reason why it does not work well with a zoom because at the beginning I save the size of every item but when the size changes, I don't get any notification about that.
I think that might be a feature in the future like your other request having a two dimensional scrolling. I think this evening I have some time to think about a solution and if it's easy to implement or not. 
But thank you already for testing these things, that helps me a lot! :) @tsquillario

**Comment by tsquillario** [comment]
Found this, seems related 

https://github.com/flutter/flutter/pull/73143

**Comment by tsquillario** [comment]
Realizing even if you can figure out how to get `InteractiveViewer` working it's probably going to be a lot of work and will take a while. I need a short term solution so I decided to hold off on `InteractiveViewer`. I can get away with vertical scrolling only (for use when in landscape orientation) and leave the GridView to fit to the screen width. I can readjust the fontSize accordingly when flipping from portrait to landscape. 

I'm experiencing 2 issues here:
1. When you scroll the GridView down in landscape orientation on a phone emulator the drag and drop is incorrectly offset. Similar to the original issue. 
2. The `enableScrollingWhileDragging` isn't working

Maybe I'm doing something wrong here....

Updated the `master` branch to reproduce the issue.
https://github.com/tsquillario/flutter-reorderable-grid-view-example1

![untitled3](https://user-images.githubusercontent.com/333300/229557506-092ce76a-4f57-4c48-9ce3-90025c372899.gif)

**Comment by tsquillario** [comment]
1. Was fixed by adding a `ScrollController`. 
2. Still an issue

**Comment by tsquillario** [comment]
Update. When I remove `ConstrainedBox` `enableScrollingWhileDragging` works but the drag offset is off like in 1.

**Comment by karvulf** [comment]
Thank you for testing so much, tomorrow I will also investigate that and come back if I have some news for you :) @tsquillario

**Comment by karvulf** [comment]
So one issue was that the widget you return doesn't have a key because usually I would expect a GridView with a key. If you add the key to the Padding, the scrolling works but there seems to be a bug because the position changes are incorrectly. @tsquillario

**Comment by karvulf** [comment]
Update: I fixed your issue, I did something wrong using the scrolling values and always calculated the positions in both directions with the same values. That leaded to wrong values in one direction.
I faced another issue, that items aren't correctly visible, so I will try to fix that too and will release an update after that. @tsquillario

**Comment by karvulf** [comment]
I released version `5.0.0-dev.5` that should fix your issue. Make sure that the child, you return in `builder` has a key that you declared at the top. @tsquillario

**Comment by tsquillario** [comment]
Sweet, works great now! I'll leave this open and update the issue to be specific to `InteractiveViewer` support. That would be a nice feature but low on the priority list. None of the other reordable gridview packages support it either.

**Comment by tsquillario** [comment]
Well, we're so close! While implementing the changes in my app I have the `GridView` inside a `Column` as a child of a `SingleChildScrollView`. This breaks the drag auto scroll.

Updated the example
https://github.com/tsquillario/flutter-reorderable-grid-view-example1

**Comment by karvulf** [comment]
Ill check that on Sunday 👍 @tsquillario
